### ISSUE TITLE
[2.4] atalkd doesn't work on Alpine when built with meson

### ISSUE THREAD

**Issue by rdmark** [Netatalk/netatalk#917]
When launching atalkd it just attempts to start routing but immediately gives up:

```
alpine:~/netatalk2$ doas ./build/etc/atalkd/atalkd -d
lo0 -phase 2 -net 0 -addr 0.0 -caddr 0.0
eth0 -phase 2 -net 65280-65534 -addr 0.0 -caddr 65280.0
```

The logs says:

```
Apr 28 17:45:00 alpine authpriv.info : dmark ran command ./build/etc/atalkd/atalkd -d as root from /home/dmark/netatalk2
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: Set syslog logging to level: LOG_DEBUG
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: restart (2.4.0dev)
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: rtmp: service unknown
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: nbp: service unknown
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: echo: service unknown
Apr 28 17:45:00 alpine daemon.info atalkd[6955]: zip: service unknown
Apr 28 17:45:00 alpine daemon.err atalkd[6955]: setifaddr: lo0 (0-0): No such device. try specifying a smaller net range.
Apr 28 17:45:00 alpine daemon.err atalkd[6955]: difaddr(0.0): No such device
```

For comparison, this is when atalkd built with autotools starts up (same system):

```
alpine:~/netatalk2$ doas ./etc/atalkd/atalkd -d
lo -phase 2 -net 0 -addr 0.0 -caddr 0.0
eth0 -phase 2 -net 65280-65534 -addr 0.0 -caddr 65280.0
packet from 65280.67 on eth0 (812) 6 (9)
06 05 00 00 00 00 00 00                         	........        
(end)
packet from 65280.67 on eth0 (812) 6 (9)
06 05 00 00 00 00 00 00                         	........        
(end)
```

```
Apr 28 17:52:37 alpine authpriv.info : dmark ran command ./etc/atalkd/atalkd -d as root from /home/dmark/netatalk2
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: Set syslog logging to level: LOG_DEBUG
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: restart (2.4.0dev)
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: rtmp: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: nbp: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: echo: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: zip: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: rtmp: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: nbp: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: echo: service unknown
Apr 28 17:52:37 alpine daemon.info atalkd[27810]: zip: service unknown
Apr 28 17:52:39 alpine daemon.info atalkd[27810]: zip_getnetinfo for eth0
Apr 28 17:52:49 alpine daemon.info atalkd[27810]: zip_getnetinfo for eth0
```

**Comment by dgsga** [comment]
@rdmark, your eagle eye on this one would be much appreciated

**Comment by NJRoadfan** [comment]
The "service unknown" errors are because entries are missing from `/etc/services`. They should be copied from the file `/config/services.atalk` from the source tree. The last time I saw that error from atalkd, it was because of the "firstnet" bug that has long since been fixed. Normally the loopback device (lo0) is not specified in atalkd.conf. Maybe that is tripping stuff up?

**Comment by dgsga** [comment]
@NJRoadfan, thanks for the tip. I checked the /etc/services file on my Alpine Linux VM, and the DDP entries were already present. I'm not really sure where to go from here... 🤔

**Comment by NJRoadfan** [comment]
OK, I played with this and found the problem, but not the cause. The reason atalkd is failing to load is that it is looking for a loop back interface named `lo0` instead of `lo` (which is correct for Alpine Linux) and failing. Adding another interface with `ip link add name lo0 type dummy` fixes this and allows atalkd to run. Somewhere a variable for the default system loopback interface is set wrong for meson vs. autotools.

**Comment by NJRoadfan** [comment]
Fixed it. in https://github.com/Netatalk/netatalk/blob/branch-netatalk-2-4/etc/atalkd/interface.h change line 48 from `#ifdef linux` to `#ifdef __linux__`

**Comment by dgsga** [comment]
Nice one! 👍

**Comment by rdmark** [comment]
Well done! There are 8 instances of the former macro in 2.4. Let's update all of them.

**Comment by dgsga** [comment]
Fixed in PR #921

**Comment by rdmark** [comment]
> The "service unknown" errors are because entries are missing from `/etc/services`. They should be copied from the file `/config/services.atalk` from the source tree. The last time I saw that error from atalkd, it was because of the "firstnet" bug that has long since been fixed. Normally the loopback device (lo0) is not specified in atalkd.conf. Maybe that is tripping stuff up?

Would it be advisable to "install" the services (modifying /etc/services ) through the build system?
